deploy/template.yaml

```diff
@@ -290,6 +295,103 @@ Resources:
+          - DNSSUFFIX:
+              !FindInMap [ EnvironmentVariables, !Ref Environment, DNSSUFFIX ]
+        - !Sub
+          - "api.${DNSSUFFIX}"
+          - DNSSUFFIX:
+             !FindInMap [ EnvironmentVariables, !Ref Environment, DNSSUFFIX ]
+      EndpointConfiguration:
+        Types:
+          - REGIONAL
+      RegionalCertificateArn: !Sub "{{resolve:ssm:/${Environment}/Platform/ACM/PrimaryZoneWildcardCertificateARN}}"
+      SecurityPolicy: TLS_1_2
+
+  CICProdApiCustomDomainName:
+    Type: AWS::ApiGateway::DomainName
+    Condition: CreateProdResources
+    Properties:
+      DomainName: 
+        - !Sub
+          - "api.${DNSSUFFIX}"
+          - DNSSUFFIX:
+             !FindInMap [ EnvironmentVariables, !Ref Environment, DNSSUFFIX ]
+      EndpointConfiguration:
+        Types:
+          - REGIONAL
+      RegionalCertificateArn: !Sub "{{resolve:ssm:/${Environment}/Platform/ACM/PrimaryZoneWildcardCertificateARN}}"
+      SecurityPolicy: TLS_1_2
+
+  CICApiCertificateRecord:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      Name: !Ref CICApiCustomDomainName
+      Type: A
+      HostedZoneId: !Sub "{{resolve:ssm:/${Environment}/Platform/Route53/PrimaryZoneID}}"
+      AliasTarget:
+        DNSName: !GetAtt CICApiCustomDomainName.RegionalDomainName
+        HostedZoneId: !GetAtt CICApiCustomDomainName.RegionalHostedZoneId
+
+  CICApiMapping:
+    Type: AWS::ApiGateway::BasePathMapping
+    DependsOn: "CICApiCustomDomainName"
+    Properties:
+      DomainName: !Ref CICApiCustomDomainName
+      RestApiId: !Ref CICRestApi
+      Stage: !Sub "${CICRestApi.Stage}"
+
+  ### End of API Gateway Custom Domain definition
+
+  ### Stub Custom Domain definition
+
+  StubApiCustomDomainName:
+    Type: AWS::ApiGateway::DomainName
+    Condition: CreateStub
+    Properties:
+      DomainName: 
+        - !Sub
+          - "ipvstub.${DNSSUFFIX}"
+          - DNSSUFFIX:
+             !FindInMap [ EnvironmentVariables, !Ref Environment, DNSSUFFIX ]
+      EndpointConfiguration:
+        Types:
+          - REGIONAL
+      RegionalCertificateArn: !Sub "{{resolve:ssm:/${Environment}/Platform/ACM/PrimaryZoneWildcardCertificateARN}}"
+      SecurityPolicy: TLS_1_2
+
+  StubApiRecord:
+    Type: AWS::Route53::RecordSet
+    Condition: CreateStub
+    Properties:
+      Name: !Ref StubApiCustomDomainName
+      Type: A
+      HostedZoneId: !Sub "{{resolve:ssm:/${Environment}/Platform/Route53/PrimaryZoneID}}"
+      AliasTarget:
+        DNSName: !GetAtt StubApiCustomDomainName.RegionalDomainName
+        HostedZoneId: !GetAtt StubApiCustomDomainName.RegionalHostedZoneId
+        EvaluateTargetHealth: false
+
+  StubApiMapping:
+    Type: AWS::ApiGateway::BasePathMapping
+    DependsOn: "StubApiCustomDomainName"
+    Condition: CreateStub
+    Properties:
+      DomainName: !Ref StubApiCustomDomainName
+      ApiId:
+        Fn::ImportValue: cic-cri-IPVStubApiGw
+      Stage: !Ref Environment
+
+  ### End of Custom Domain definition
```

Comment:
should these be in the stub template, rather than the main deploy template?

Reply:
now moved to stub template